Source/Android/jni/MainAndroid.cpp

```diff
@@ -161,12 +161,12 @@ static bool MsgAlert(const char* caption, const char* text, bool yes_no, MsgType
   g_java_vm->AttachCurrentThread(&env, NULL);
 
   // Execute the Java method.
-  env->CallStaticVoidMethod(s_jni_class, s_jni_method_alert, env->NewStringUTF(text));
+  jboolean result = env->CallStaticBooleanMethod(s_jni_class, s_jni_method_alert, env->NewStringUTF(caption), env->NewStringUTF(text), yes_no ? JNI_TRUE : JNI_FALSE);
 
   // Must be called before the current thread exits; might as well do it here.
   g_java_vm->DetachCurrentThread();
 
-  return false;
+  return (bool)(result != JNI_FALSE);
```

Comment:
Doesn't simply `return result != JNI_FALSE` work fine?

Reply:
Was apparently asleep when writing that. Fixed.